locust/clients.py

```diff
@@ -187,6 +241,34 @@ def _send_request_safe_mode(self, method, url, **kwargs):
             r.request = Request(method, url).prepare()
             return r
 
+    # These # type: ignore[override] comments below are needed because our overridden version of functions receives
+    # more arguments than functions in the base class.
+    def get(self, url: str | bytes, **kwargs: Unpack[RESTKwargs]):  # type: ignore[override]
+        kwargs.setdefault("allow_redirects", True)
+        return self.request("GET", url, **kwargs)  # type: ignore[misc]
+
+    def options(self, url: str | bytes, **kwargs: Unpack[RESTKwargs]):  # type: ignore[override]
+        kwargs.setdefault("allow_redirects", True)
+        return self.request("OPTIONS", url, **kwargs)  # type: ignore[misc]
+
+    def head(self, url: str | bytes, **kwargs: Unpack[RESTKwargs]):  # type: ignore[override]
+        kwargs.setdefault("allow_redirects", False)
+        return self.request("HEAD", url, **kwargs)  # type: ignore[misc]
+
+    # These # type: ignore[misc] comments below are needed because data and json parameters are already defined in the
+    # RESTKwargs TypedDict. An alternative approach is to define another TypedDict which doesn't contain them.
+    def post(self, url: str | bytes, data: Any | None = None, json: Any | None = None, **kwargs: Unpack[RESTKwargs]):  # type: ignore[override, misc]
```

Comment:
There does seem to be a problem when using pyright after all. For HTTP methods where  `data=data` and/or `json=json` is passed to `self.request(...)` the return type becomes `Unknown`
![image](https://github.com/locustio/locust/assets/28349560/720c3a67-4d09-497c-a180-f15e7b00a2ea)

I'm unsure what causes this behaviour. Explicitly providing the return type seems to fix it (see suggested changes).

```suggestion
    def post(self, url: str | bytes, data: Any | None = None, json: Any | None = None, **kwargs: Unpack[RESTKwargs]) -> (ResponseContextManager | Response | LocustResponse):  # type: ignore[override, misc]
```

Now the correct return type is shown:
![image](https://github.com/locustio/locust/assets/28349560/25b25044-7a6a-4a56-ac73-fb780c13105a)